src/Chainweb/Pact/Service/Types.hs

```diff
@@ -168,6 +170,10 @@ data PactException
   | BuyGasFailure !GasPurchaseFailure
   | MempoolFillFailure !Text
   | BlockGasLimitExceeded !Gas
+  | LocalRewindLimitExceeded
+    { _localRewindExceededLimit :: !Natural
+    , _localRewindRequestedDepth :: !BlockHeight }
```

Comment:
Nit, I think either both of these should be `BlockHeight` or both of these should be `Natural`, because they're both the same "type of thing" (block height delta)

Reply:
I'm being consistent here with `RewindLimitExceeded`, where the limit is `Natural` and heights are `BlockHeight`. 

And semantics is a bit different — limit says how far in the past we can go, while the height starts from the bottom. Different directions.

Reply:
Aren't they both measured from the tip of the chain?

Reply:
The rewind's limit starts from the current moment, how far we can go backwards.